src/util/psbt/mod.rs

```diff
@@ -183,6 +185,103 @@ impl PartiallySignedTransaction {
 
         Ok(())
     }
+
+    // TODO: Work out attribution/license stuff, copied from `impl ComputeSighash for Legacy` in `bdk`.
+    /// Returns the sighash (and type) for input at `input_index`, caller is
+    /// responsible for ensuring `input_index` is a legacy input.
+    pub fn legacy_sighash(&self, input_index: usize) -> Result<(Sighash, EcdsaSighashType), Error> {
```

Comment:
I think we can actually do better :) . See https://docs.rs/miniscript/7.0.0-rc.2/miniscript/psbt/trait.PsbtExt.html#tymethod.sighash_msg . This figures out what the type of spent input is and applies the corresponding signature hash algorithm.

I was planning to upstream this but forgot about it.